plugins/uniques.js

```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+				confirm('Import Successful.');
+				// to write back the data to localStorage
+				window.plugin.uniques.storeLocal('uniques');
+			} catch(e) {
+				console.warn('UNIQUES: failed to import data: '+e);
+			}
+		});
+}
+
+/*****************************************************************************************/
+/** UNIQUES Backlog to be processed whenever a portal's LatLonE6 can be resolved to GUID */
+/*****************************************************************************************/
+window.plugin.uniques.onPortalAdded = function(data) {
+	var guid = data.portal.options.guid;
+	var p = data.portal.options.data;
+	var id = p.latE6 + "," + p.lngE6;
+
+	//console.log("portal added: %s %s %o", guid, id, data);
+	// check if portal is in missedLatLngs
+	if (window.plugin.uniques.missedLatLngs[id]) {
+		console.log("found portal guid for previously missed portal: %s -> %s (%o)", id, guid, data);
+		window.plugin.uniques.uniques[guid] = Object.assign (
+			{},
+			window.plugin.uniques.uniques[guid],
+			window.plugin.uniques.missedLatLngs[id].action
+		);
+		window.plugin.uniques.storeLocal('uniques');
+		delete window.plugin.uniques.missedLatLngs[id];
+		window.plugin.uniques.storeLocal('missedLatLngs');
+		$("#missedPL").html(window.plugin.uniques.genList());
+	}
+}
+
+window.plugin.uniques.removeOldParsedMsgs = function() {
+//	remove all timestamps older than 30 days
+	var old = (Date.now() - (30*24*60*60*1000)); //30 days in miliseconds
+	var count = 0;
+	for (var item in window.plugin.uniques.parsedMsgs){
+		if (item < old) {
+			delete item;
+			count++;
+		}
+	}
+	console.log ('[uniques] removed %s old parsedMsgs', count);
+}
+
+window.plugin.uniques.genList = function (){
+	var mLL = window.plugin.uniques.missedLatLngs;
+	var list = '';
+	for (var item in mLL) {
+		var p = mLL[item].portal;
+		list = list + '<a onclick=\"map.setView(['+p.latE6/1E6+','+p.lngE6/1E6+'],15);\">'+p.name+'</a><br>';
+	}
+	return list
+}
 
+window.plugin.uniques.options = function (){
+	aoPortals=window.plugin.uniques.uniques;
+	visited=captured=scouted=droned=0;
+	$.each(aoPortals,function(PUID){
+		aPortal=window.plugin.uniques.uniques[PUID];
+		if (aPortal.visited) visited++;
+		if (aPortal.captured) captured++;
+		if (aPortal.scouted) scouted++;
+		if (aPortal.droned) droned++;
+	});
 
+	let list = 'Unique Portals Count:<br>visited: '+visited+' - captured: '+captured+'<br>scouted: '+scouted+' - droned: '+droned+'<br><hr>'
+			+ 'Missed Portals: <br>Click on portals to move the map to resolve the backlogged actions for this portal.<br><br>'
+			+ '<div id="missedPL" style="height:150px;overflow:auto">'
+			+ window.plugin.uniques.genList()
+			+ '</div><hr>'
+			+ '<a onclick=\"window.plugin.uniques.optExport();return false\" title=\"Export portals\' unique info to IITC.\">Backup</a> / '
+			+ '<a onclick=\"window.plugin.uniques.optImport();return false\" title=\"Import portals\' unique info to IITC.\">Restore</a> Uniques'
```

Comment:
```suggestion
			+ '<a onclick="window.plugin.uniques.optExport();return false" title="Export portals\' unique info to IITC.">Backup</a> / '
			+ '<a onclick="window.plugin.uniques.optImport();return false" title="Import portals\' unique info to IITC.">Restore</a> Uniques'
```
And is it possible to replace inline js with jQuery listeners?

Reply:
changed